src/LinuxTracing/ThreadDepedencyTestApp.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <absl/strings/str_format.h>
+#include <absl/synchronization/mutex.h>
+
+#include <atomic>
+#include <chrono>
+#include <csignal>
+#include <iostream>
+#include <iterator>
+#include <mutex>
+#include <thread>
+
+#include "OrbitBase/ThreadUtils.h"
+
+static std::atomic<bool> exit_requested;
+static std::atomic<int> num;
+
+absl::Mutex test_mutex;
+
+constexpr int kNumOfThreads = 4;
+constexpr int kSigint = 2;
+
+std::thread my_threads[kNumOfThreads];
+
+void SignalHandler(int signum) {
+  std::cout << "Cleaning up!" << std::endl;
+  exit_requested.store(false);
+  num++;
+  for (auto& my_thread : my_threads) {
+    my_thread.join();
+  }
+  exit(signum);
+}
+
+static void DoWork(int thread_num) {
+  std::string thread_name = absl::StrFormat("Thread %d", thread_num);
+  orbit_base::SetCurrentThreadName(thread_name.c_str());
+  while (true) {
+    if (!exit_requested.load()) {
+      break;
+    }
+    int temp = num.load();
+    absl::MutexLock test_mutex_lock(&test_mutex);
+    while (temp == num.load())
+      ;
+  }
+}
+
+int main() {
+  // This program just spawns kNumOfThreads threads and has a shared mutex between all of them.
```

Comment:
In the screenshot you uploaded, both Thread3 and Thread2 are sleeping for a while. Is there any guarantee that they eventually will wake up and you will have exactly kNumOfThreads fighting for run?

Reply:
Yes they do. The screenshot I posted was extremely zoomed in.

Should I change it?

Reply:
Nono. The screenshot is perfect. I just was a bit cryptid for the behavior of the threads and why some of them were grey at that time but only one woke up but I might be missing something. Anyway, don't worry too much.